library/kani/src/futures.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, RawWaker, RawWakerVTable, Waker},
+};
+
+/// Polls the future in a loop until completion (a very simple executor)
+///
+/// This is intended as a drop-in replacement for `futures::block_on`, which Kani cannot handle.
+pub fn block_on<T>(mut fut: impl Future<Output = T>) -> T {
+    let waker = unsafe { Waker::from_raw(NOOP_RAW_WAKER) };
+    let cx = &mut Context::from_waker(&waker);
+    // SAFETY: we shadow the original binding, so it cannot be accessed again for the rest of the scope.
+    // This is the same as what the pin_mut! macro in the futures crate does.
+    let mut fut = unsafe { Pin::new_unchecked(&mut fut) };
+    loop {
+        match fut.as_mut().poll(cx) {
+            std::task::Poll::Ready(res) => return res,
+            std::task::Poll::Pending => continue,
+        }
+    }
+}
+
+/// A dummy waker, which is needed to call [`Future::poll`]
+const NOOP_RAW_WAKER: RawWaker = {
```

Comment:
Can this be overriden?

Reply:
I'm not sure what you mean? It's a constant, so it cannot be changed.

Reply:
My question was really "should there be a way to override this"

Reply:
I don't foresee a way to take advantage of the waker for now in the context of Kani. Normal execution makes use of threads that are woken up by the OS (if I understand correctly). But that is not an option for Kani. In any case, if we add another executor, it will need a different name than `block_on` and will be able to use a different waker, if necessary.